### ISSUE TITLE
Change of scaling without log out/log in

### ISSUE THREAD

**Issue by kalj** [fuhsjr00/bug.n#155]
Currently, it is necessary to log out of Windows and back in again for bug.n to correctly work after a change of scaling. This is problematic when logging into a live session remotely from a laptop with scaling enabled, since the destination computer will then adapt its scaling to the laptop. It is not a practical option to log in/out since the computer is a workstation with many programs running (which was facilitated by bug.n in the first place ;) ). Thus, I would like to request support for scaling change without having to sign in/out.

This is related to #18.

**Comment by kalj** [comment]
I can add there is the (quite suboptimal) workaround of disabling scaling on the laptop, and instead setting a lower screen resolution. It is weird that change of resolution works smoothly without a relogin, where scaling doesn't. It would still be nice to see this fixed.

**Comment by joten** [comment]
Does reloading bug.n with <kbd>Win</kbd><kbd>Ctrl</kbd><kbd>R</kbd> do any good?

**Comment by jloow** [comment]
I'm also experiencing scaling problems. For me they occur when I remove my laptop from it's dock or when I plug it into the dock again. In "laptop mode" the scaling is set to 150%, while in the dock it's 100%. So for example, I will be working from the dock using bug.n, but if I undock the computer the scaling of the status bar gets messed up. Similarly, if i started bug.n in laptop mode and then dock the computer, the scaling of the status bar will also be wrong.

Reloading bug.n with <kbd>Win</kbd><kbd>Ctrl</kbd><kbd>R</kbd> does not solve the problem, neither does restarting the program. As far as I can tell only signing in and out will work.

After some quick googling I suspect the problem is with AHK; there are several posts reporting scaling problems for applications based on AHK.

**Comment by joten** [comment]
I tried to reproduce the problem, but could not do so. I am currently using a chromebook with display scaling and connected to a Windows 10 via RDP.

I looked up the code regarding scaling and it looks a bit dated. It is reading the DPI scaling from the registry, which would explain the need for a re-login. Does setting it manually and restarting bug.n have any effect? It might be possible to do somethig with `A_ScreenDPI`, but I cannot really test it.

**Comment by jloow** [comment]
I tried changing the scaling manually by editing `LogPixels` and `Win8DpiScaling` under `
HKEY_CURRENT_USER\Control Panel\Desktop`, but this does not work (unless I sign in and out).

As an aside: while testing this, I also noticed that when bug.n gives you the option to reconfigure the monitors, and you press 'yes', this does temporarily fix the scaling issue with the status bar. However, the windows will not scale properly. When you reload bug.n, the status bar gets scaled wrong again but the windows are correct.

**Comment by joten** [comment]
Does setting e.g. `Config_scalingFactor=0.75` (a value according to your display settings) in Config.ini have any effect?

**Comment by jloow** [comment]
Unfortunately, that did not work either. 

Setting `Config_barWidth=70%` does make the whole status bar appear on the screen, but the spacing between the bar and the windows is smaller than the other spacings (e.g. between two windows). It is, however, the solution I'm currently using.

**Comment by joten** [comment]
With commit https://github.com/fuhsjr00/bug.n/commit/413a7c19760eec6c511ab938cc9cf295847c881d to the X branch I added per-monitor display scaling awareness. I could only test in a single monitor system, but could reproduce the issues regarding a too wide bug.n status bar and layout at first, and the changes had at least a mitigating effect. A logout is not necessary, but you will have to restart bug.n, although resetting the monitor (if asked) does work, but leaves an untidy layout. Have a try, if you have time.

**Comment by jloow** [comment]
Yes, that did work! I don't even have to restart bug.n; I only have to reload it using <kbd>Win</kbd><kbd>Ctrl</kbd><kbd>R</kbd>. Just resetting the monitor when queried works as well (and doesn't leave an untidy layout for me).

Thanks!

**Comment by joten** [comment]
With commit 0f9f537040bdca074feb52226050e3c680c3003c I also added the changes to the master branch; the X branch will get a bit more unstable.

If it is sufficiently tested, I will release it with version 9.1.0, therewith providing an executable.
